lib/Mojo/Promise.pm

```diff
@@ -63,6 +63,12 @@ sub then {
   return $new;
 }
 
+sub timeout {
+  my ($self, $after, $err) = (blessed $_[0] ? shift->clone : shift->new, @_);
```

Comment:
Other methods use `ref` instead of `blessed`.

Reply:
The use of `clone` here is actually inconsistent (and there appears to be no test for it).

Reply:
There should also be a default error message, so `Mojo::Promise->timeout(5)` works too.

Reply:
What do you want the default error message to be? "Timeout"..?

Reply:
"Promise timeout" would seem reasonable, no?